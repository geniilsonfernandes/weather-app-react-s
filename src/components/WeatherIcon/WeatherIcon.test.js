import { render, screen } from "@testing-library/react";
import { WeatherIcon } from "./index";

describe("<WeatherIcon />", () => {
  it("shoud icon render with label and title", () => {
    const { container } = render(
      <WeatherIcon iconCode={200} label="clear sky" />
    );
    expect(screen.getByTitle("clear sky")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it("shoud icon no found when icon no pass", () => {
    const { container } = render(<WeatherIcon />);
    expect(screen.getByTitle("no found")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
