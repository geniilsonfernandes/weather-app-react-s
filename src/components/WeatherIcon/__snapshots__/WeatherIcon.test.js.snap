// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherIcon /> shoud icon no found when icon no pass 1`] = `
<div>
  <svg
    aria-label="no found"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      no found
    </title>
    <g>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12.866 3l9.526 16.5a1 1 0 0 1-.866 1.5H2.474a1 1 0 0 1-.866-1.5L11.134 3a1 1 0 0 1 1.732 0zM11 16v2h2v-2h-2zm0-7v5h2V9h-2z"
      />
    </g>
  </svg>
</div>
`;

exports[`<WeatherIcon /> shoud icon render with label and title 1`] = `
<div>
  <i
    alt="clear sky"
    class="sc-bczRLJ owf owf-200"
    title="clear sky"
  />
</div>
`;
